### ISSUE TITLE
Docstring rendering doesn't handle indentation

### ISSUE THREAD

**Issue by Tazarus** [microsoft/pylance-release#5446]
Type: <b>Bug</b>

## Expected Behaviour: 

When viewing a docstring via the tooltip it should have the same formatting as a docstring produced by the help() function or called via the .__doc__ attribute.

## Actual Behaviour: 

The tooltip does not parse newlines and indentations in the same way, this causes tooltips and docstrings to require different formatting to be easily legible in VScode vs Python help.

## Steps to reproduce: 

Write any function that uses multiple levels of indentation with multiline strings in the docstring. Then call the help() function and hover over the tooltip and view the differences.

```
def format_example_func(a:str, b:int):
    """This function made expressly to show docstring formatting and nothing else
    
    For some reason the parsing has a really hard time when I write a docstring
    where both multiline docstrings and indentation exist simultaneously.

        Params:
            a: a string you want to do something with, who knows what it actually
               does this is just an example so people can call the help function
            b: another random arguement intended to showcase what changes when you
               combine indentation with multiline docstrings

        Returns:
            foo: I don't even know what this thing will return blah blah thingy
                 stuff stuff stuff.
    """

    return a*b

print(help(format_example_func))
```

![Docstring formatting](https://github.com/microsoft/vscode-python/assets/69982590/4c828b17-268e-4ac1-9c4a-ead0e906eec1)
![tooltip1](https://github.com/microsoft/vscode-python/assets/69982590/1e0c328e-b63f-4039-befe-88449185f4ed)
![tooltip2](https://github.com/microsoft/vscode-python/assets/69982590/1d8b325e-4b50-463b-86f1-cb892df758ea)

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.5
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Conda
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

</p>
</details>

<details>

<summary>User Settings</summary>

</p>

```

languageServer: "Pylance"

terminal
• executeInFileDir: "<placeholder>"

```

</p>
</details>

Extension version: 2024.0.0
VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz (4 x 3192)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|7.88GB (1.74GB free)|
|Process Argv|--crash-reporter-id a03e095a-dadf-4638-8b94-73ea16da42e2|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30951796
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019bcf:30953938
3ef8e399:30949928

```

</details>

**Comment by rchiodo** [comment]
Thanks for the issue.  Seems like generally the indentation doesn't work very well.

**Comment by luabud** [comment]
Dupe of https://github.com/microsoft/pylance-release/issues/3347?

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2024.6.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20246100-13-June-2024-PreRelease)

**Comment by rchiodo** [comment]
The fix for this issue is behind an experimental feature flag:

```json
"python.analysis.supportRestructuredText": true
```

If you want to try our our new restructuredText support, enable this flag. It's behind a flag at the moment until we can make sure it handles all the possible Sphinx/GoogleDoc/Epytext scenarios that customers need. Please log additional issues if the flag is not meeting your needs.

Thanks :)
